### ISSUE TITLE
lag.xts segfault on empty xts object with dims

### ISSUE THREAD

**Issue by joshuaulrich** [joshuaulrich/xts#65]
```r
x <- xts(,Sys.time())  # empty xts object
lag(x)  # throws uninformative error

# but if x has a non-NULL dim attribute with zero columns...
dim(x) <- c(1,0)
lag(x)  # segfault
```

**Comment by joshuaulrich** [comment]
This is actually a bug in the lag C code that has been moved upstream into zoo, and I fixed it in zoo in [revision 986](https://r-forge.r-project.org/scm/viewvc.php/pkg/zoo/src/lag.c?root=zoo&r1=937&r2=986&pathrev=986) in May, 2014.  I'll ping the zoo maintainer to get an ETA on getting an updated zoo to CRAN.

**Comment by joshuaulrich** [comment]
Closing, since this is fixed in zoo 1.7-12.

```r
R> suppressPackageStartupMessages(require(xts))
R> packageVersion("zoo")
[1] ‘1.7.12’
R> x <- xts(,Sys.time())  # empty xts object
R> lag(x)  # throws informative error
Error in lag.xts(x) : abs(k) must be less than nrow(x)
R> dim(x) <- c(1,0)
R> lag(x)  # no segfault
                   
2015-03-21 20:46:43
R> identical(x,lag(x))
[1] TRUE
```
